lib/mongo/socket/ssl.rb

```diff
@@ -137,7 +137,7 @@ def initialize(host, port, host_name, timeout, family, options = {})
       #
       # @since 2.0.0
       def connect!
-        Timeout.timeout(options[:connect_timeout], Error::SocketTimeoutError) do
+        Timeout.timeout(options[:connect_timeout], Error::SocketTimeoutError, 'The socket took over #{options[:connect_timeout]} seconds to connect') do
```

Comment:
The string needs double quotes to perform interpolation.

You can test the message that is generated by adding e.g. a `sleep 10` call in the block here, and instantiating a client with a lower connect timeout: `Mongo::Client.new(['localhost'], connect_timeout: 1)`. This should raise the timeout error with the message you are adding here.

Reply:
fixed and tested.